go/test/endtoend/mysqlctld/mysqlctld_test.go

```diff
@@ -134,10 +135,29 @@ func initCluster(shardNames []string, totalTabletsRequired int) error {
 	return nil
 }
 
+const defaultOperationTimeout = 60 * time.Second
+const defeaultRetryDelay = 3 * time.Second
+
+func waitForMysqlctldShutdown(t *testing.T, tab *cluster.Vttablet) bool {
+	tmr := time.NewTimer(defaultOperationTimeout)
+	for {
+		if tab.MysqlctldProcess.HasShutdown() {
+			return true
+		}
+		select {
+		case <-tmr.C:
+			return false
+		default:
+		}
+		time.Sleep(defeaultRetryDelay)
```

Comment:
We're only checking if the variable is nil. Why wait 3 seconds?

Reply:
See comment below:  there is a process wait in `Start()` waiting for process to end (triggered by `Stop()`). The delay is what is causing the flakiness.